ext/mochilo/buffer.c

```diff
@@ -95,15 +95,20 @@ VALUE mochilo_buf_flush(mochilo_buf *buf)
 
 mochilo_buf_chunk *mochilo_buf_rechunk2(mochilo_buf *buf, size_t chunk_size)
 {
+	mochilo_buf_chunk *chunks;
+
 	skip_last_chunk(buf);
 
 	if (buf->cur_chunk == buf->chunk_count) {
-		buf->chunk_count *= 2;
+		if (!(buf->chunk_count * 2))
```

Comment:
This test assumes that `buf->chunk_count` is a power of two.  This is true today, but I don't know if the design requires it forever.

Something like `if ( (buf->chunk_count * 2) < buf->chunk_count)` might be more robust, if you think it is necessary.